### PR TITLE
basic organization permission system

### PR DESCRIPTION
**Pull request by Rizziepit**
- [x] views must only show and allow editing of projects for current organization
- [x] views must check individual and group permissions
- [x] developing new views with perms + organization scope must be easy

### CODE REVIEW

**Comment by Rizziepit** [comment]
@miltontony @smn ready for review. This adds permissions and groups to user-organization relations. It also applies permissions and filtering to views. So when this lands we won't have a global view of all projects anymore (aside from the admin views).

**Comment by miltontony** [comment]
Should we not have an "All" option for special users or a special group perhaps? It's definitely still needed.

**Comment by Rizziepit** [comment]
I can add an "All" option for staff users. I''ll have to update view logic a bit. Can I do it in another PR that lands along with this one?

**Comment by miltontony** [comment]
Yes, another PR should be fine

File: unicoremc/views.py [hunk 1]
```diff
@@ -54,8 +57,31 @@ def get_all_repos(request):
     return HttpResponse(json.dumps(repos), content_type='application/json')
 
 
+class ProjectViewMixin(View):
```
Comment (miltontony) [inline unicoremc/views.py:60]: :star: :star2: really like this implementation

**Comment by miltontony** [comment]
:+1: :cake: :cake:

File: unicoremc/views.py (hunk 1 above)
Comment (Rizziepit) [inline unicoremc/views.py:60]: Yay!

**Pull request merged**